calculations/temperature_calculations.py

```diff
@@ -1,4 +1,57 @@
 class TemperatureCalculator:
+    
+    def compile_data_for_yearly_calculations(self, weather_readings):
+        minimum_temperature_every_month = {}
+        maximum_temperature_every_month = {}
+        maximum_humidity_every_month = {}
+        
+        for year_number, year_values in weather_readings.items():
+            minimum_temperature = []
+            maximum_temperature = []
+            maximum_humidity = []
+            
+            for month_name, month_values in year_values.items():
+                minimum_temperature.append(
+                    min(month_values, key=lambda x: int(x['Min TemperatureC'])))
+                minimum_temperature_every_month[year_number] = minimum_temperature
+                maximum_temperature.append(
+                    max(month_values, key=lambda x: int(x['Max TemperatureC'])))
+                maximum_temperature_every_month[year_number] = maximum_temperature
+                maximum_humidity.append(
+                    max(month_values, key=lambda x: int(x[' Mean Humidity'])))
+                maximum_humidity_every_month[year_number] = maximum_humidity
+                
+        return {'minimum_temperature_every_month': minimum_temperature_every_month, 
+                'maximum_temperature_every_month': maximum_temperature_every_month, 
+                'maximum_humidity_every_month': maximum_humidity_every_month}
+
+
+    def compile_data_for_monthly_calculations(self, weather_readings):
+        avg_minimum_temperature_every_month = {}
+        avg_maximum_temperature_every_month = {}
+        avg_mean_humidity_every_month = {}
+        
+        for key_year, year in weather_readings.items():
+            for month_name, month_values in year.items():
+                
+                avg_maximum_temperature_every_month.setdefault(
+                    key_year, {}).setdefault(month_name, float(
+                        sum(map(int, (daily_readings['Max TemperatureC'] for daily_readings in month_values 
+                                      if daily_readings['Max TemperatureC'] != '')))) / len(month_values))
```

Comment:
Readability is quite compromised, can you make ii readable by formatting well?

Reply:
mapping to int was not required as we had already type casted while reading files.
Does following formatting justify readability?

'            for month_name, month_values in year.items():
                
                avg_maximum_temperature_every_month.setdefault(key_year, {}).setdefault(month_name, float(
                    sum(daily_readings['Max TemperatureC'] for daily_readings in month_values
                        if daily_readings['Max TemperatureC'] != '') // len(month_values)))
                
                avg_minimum_temperature_every_month.setdefault(key_year, {}).setdefault(month_name, float(
                    sum(daily_readings['Min TemperatureC'] for daily_readings in month_values
                        if daily_readings['Min TemperatureC'] != '') // len(month_values)))
                
                avg_mean_humidity_every_month.setdefault(key_year, {}).setdefault(month_name, float(
                    sum(daily_readings[' Mean Humidity'] for daily_readings in month_values
                        if daily_readings[' Mean Humidity'] != '') // len(month_values)))'